DocumentsFromSnapshotMigration/src/test/java/com/rfs/EndToEndTest.java

```diff
@@ -0,0 +1,277 @@
+package com.rfs;
+
+import java.io.File;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+import org.opensearch.migrations.metadata.tracing.MetadataMigrationTestContext;
+import org.opensearch.migrations.reindexer.tracing.DocumentMigrationTestContext;
+import org.opensearch.migrations.snapshot.creation.tracing.SnapshotTestContext;
+import org.opensearch.migrations.workcoordination.tracing.WorkCoordinationTestContext;
+
+import com.rfs.common.FileSystemRepo;
+import com.rfs.common.FileSystemSnapshotCreator;
+import com.rfs.common.OpenSearchClient;
+import com.rfs.common.http.ConnectionContextTestParams;
+import com.rfs.framework.SearchClusterContainer;
+import com.rfs.http.ClusterOperations;
+import com.rfs.worker.SnapshotRunner;
+import lombok.Lombok;
+import lombok.SneakyThrows;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class EndToEndTest extends SourceTestBase {
+    @TempDir
+    private File localDirectory;
+
+    @ParameterizedTest(name = "Target OpenSearch {0}")
+    @ArgumentsSource(SupportedTargetCluster.class)
+    public void migrateFrom_ES_v6_8(final SearchClusterContainer.Version targetVersion) throws Exception {
+        var executorService = Executors.newFixedThreadPool(1);
+        final var snapshotContext = SnapshotTestContext.factory().noOtelTracking();
+        final var metadataContext = MetadataMigrationTestContext.factory().noOtelTracking();
+        final var workCoordinationContext = WorkCoordinationTestContext.factory().noOtelTracking();
+        final var testDocMigrationContext = DocumentMigrationTestContext.factory(workCoordinationContext).noOtelTracking();
+
+        try (
+            final var sourceCluster = new SearchClusterContainer(SearchClusterContainer.ES_V6_8_23);
+            final var targetCluster = new SearchClusterContainer(targetVersion)
+        ) {
+            // === ACTION: Set up the source/target clusters ===
+            var bothClustersStarted = CompletableFuture.allOf(
+                CompletableFuture.runAsync(() -> sourceCluster.start()),
+                CompletableFuture.runAsync(() -> targetCluster.start())
+            );
+            bothClustersStarted.join();
+
+            // Create a template
```

Comment:
Since this test is focused on the document reindexing, seems like all the template logic/metadata migration could be removed as its duplicated in `MetadataMigration/src/test/java/org/opensearch/migrations/EndToEndTest.java`.  What do you think about separating the validation of the metadata toolset from the document?

Note; if we did remove this validation from this test case, then can you make sure the ES6.8 scenarios are working in those tests cases?

Reply:
I'd like to have a end-to-end test in our unit tests, happy to have that live here or elsewhere, and have that be a simplified case.  I'm also fine having some overlap with the (see these and `ParallelDocumentMigrationsTest`, for example).  What if we keep these tests how they are and you add your own transformation tests with much more complex inputs in the `MetadataMigration` sub-project?